### PR TITLE
Mob rarities

### PR DESCRIPTION
**Pull request by funkychicken493**

### CODE REVIEW

**Comment by funkychicken493** [comment]
Prefixes:
- Cowardly: runs away when attacked
- Breaking: disables player's items on attack
- Light: opposite of heavy, gains speed on damaged
- Launching: higher knock-back
- Sandy: sand
- Drunkening: makes you more drunk on attack
- Draining: removes buffs on attack
- Spinning: makes hit entity spin
- Hungering: removes hunger from food bar instead of giving the hunger effect
- Attractive: forces players to look at it when it attacks
- Shrieking: 5 percent chance to spawn the warden on attack
- Drowning: removes oxygen
- Attaching: attaches attacker to the victim with a lead
- Confusing: shuffles player hot-bar
- Robbing: forces player to drop held item
- Rocketeering: launches players upwards with a firework


Suffixes:
- Reaping: healed when nearby mobs die
- Undying: chance to not take damage from an attack at all
- Summoning: spawns silverfish on attack
- Steelskinned: greater defense
- Teleporting: warps to target
- Splitting: splits into two upon death

File: denizen_scripts/HeroCraft/mobs/prefixes/attaching.dsc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+custom_mob_prefix_attaching:
+  Type: world
+  debug: false
+  events:
+    after entity_flagged:attaching damages entity:
```
Comment (Ph4i1ur3) [inline denizen_scripts/HeroCraft/mobs/prefixes/attaching.dsc]: I don't think this actually does anything

File: denizen_scripts/HeroCraft/mobs/prefixes/breaking.dsc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+custom_mob_prefix_breaking:
+  Type: world
+  debug: false
+  events:
+    after entity_flagged:breaking damages player:
```
Comment (Ph4i1ur3) [inline denizen_scripts/HeroCraft/mobs/prefixes/breaking.dsc:5]: probably needs a check to not disable air

File: denizen_scripts/HeroCraft/mobs/prefixes/attaching.dsc (hunk 1 above)
Comment (funkychicken493) [reply denizen_scripts/HeroCraft/mobs/prefixes/attaching.dsc]: honestly idk

File: denizen_scripts/HeroCraft/mobs/prefixes/breaking.dsc (hunk 2 above)
Comment (funkychicken493) [reply denizen_scripts/HeroCraft/mobs/prefixes/breaking.dsc:5]: 👍

File: denizen_scripts/HeroCraft/mobs/prefixes/drowning.dsc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+custom_mob_prefix_drowning:
+  type: world
+  debug: false
+  events:
+    after player damaged by entity_flagged:drowning:
```
Comment (Ph4i1ur3) [inline denizen_scripts/HeroCraft/mobs/prefixes/drowning.dsc]: This seems useless in 90% of scenarios

Comment (funkychicken493) [reply denizen_scripts/HeroCraft/mobs/prefixes/drowning.dsc]: yeah

File: denizen_scripts/HeroCraft/mobs/prefixes/shrieking.dsc [hunk 4]
```diff
@@ -0,0 +1,7 @@
+custom_mob_prefix_shrieking:
+  Type: world
+  debug: false
+  events:
+    after entity_flagged:shrieking damages entity:
+        - stop if:<util.random_chance[95]>
```
Comment (Ph4i1ur3) [inline denizen_scripts/HeroCraft/mobs/prefixes/shrieking.dsc]: 5% is REALLY HIGH, move this to a suffix

Comment (funkychicken493) [reply denizen_scripts/HeroCraft/mobs/prefixes/shrieking.dsc]: k

File: denizen_scripts/HeroCraft/mobs/suffixes/reaping.dsc [hunk 5]
```diff
@@ -0,0 +1,9 @@
+custom_mob_suffix_reaping:
+  Type: world
+  debug: false
+  events:
+    after entity dies chance:50:
+        - define reaping <context.location.find_entities[*].within[10].filter_tag[<[filter_value].has_flag[reaping]>]>
```
Comment (Ph4i1ur3) [inline denizen_scripts/HeroCraft/mobs/suffixes/reaping.dsc:6]: What applies reaping?

Comment (funkychicken493) [reply denizen_scripts/HeroCraft/mobs/suffixes/reaping.dsc:6]: oops i forgotted

Comment (funkychicken493) [reply denizen_scripts/HeroCraft/mobs/suffixes/reaping.dsc:6]: actually, what do you mean?

Comment (funkychicken493) [reply denizen_scripts/HeroCraft/mobs/suffixes/reaping.dsc:6]: @Ph4i1ur3

Comment (Ph4i1ur3) [reply denizen_scripts/HeroCraft/mobs/suffixes/reaping.dsc:6]: Oh, I see! Let's make this more fun, shall we? Increase max health proportionately. heheh

Comment (funkychicken493) [reply denizen_scripts/HeroCraft/mobs/suffixes/reaping.dsc:6]: oh god

**Comment by funkychicken493** [comment]
ggs gamers

**Comment by funkychicken493** [comment]
Mob rarities post merge:
- Cowardly doesn't work, mob ignores the walk command
- Spinning needs to have the interval decreased
- Confusing doesn't work
- Splitting doesn't work
- Sandy should send a title instead of a chat message
- Sandy should create sand particle effects
- Breaking doesn't actually break anything, you can still use a cool-downed weapon

I will resolve these issues myself.

**Pull request merged**